server/pulp/server/webservices/views/consumers.py

```diff
@@ -0,0 +1,1047 @@
+    @json_body_allow_empty
+    def put(self, request, consumer_id,  schedule_id):
+        """
+        Update a specific schedule <action>.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: The consumer ID.
+        :type consumer_id: str
+        :param schedule_id: the schedule id
+        :type schedule_id: str
+
+        :return: Response containing consumer's updated schedule <action>
+        :rtype: django.http.HttpResponse
+        """
+
+        schedule_data = request.body_as_json
+        options = schedule_data.pop('options', None)
+        units = schedule_data.pop('units', None)
+
+        if 'schedule' in schedule_data:
+            schedule_data['iso_schedule'] = schedule_data.pop('schedule')
+
+        schedule = self.manager.update_schedule(consumer_id, schedule_id, units,
+                                                options, schedule_data)
+
+        scheduled_obj = scheduled_unit_management_obj(schedule.for_display())
+        add_link_schedule(scheduled_obj, self.ACTION, consumer_id)
+        return generate_json_response_with_pulp_encoder(scheduled_obj)
+
+    @auth_required(authorization.DELETE)
+    def delete(self, request, consumer_id, schedule_id):
+        """
+        Delete a specific schedule <action>.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: The consumer ID.
+        :type consumer_id: str
+        :param schedule_id: the schedule id
+        :type schedule_id: str
+
+        :return: An empty response
+        :rtype: django.http.HttpResponse
+        """
+
+        self.manager.delete_schedule(consumer_id, schedule_id)
+        return generate_json_response(None)
+
+
+class UnitInstallScheduleView(ConsumerUnitActionScheduleView):
+    """
+    View for scheduled content install on the consumer.
+    """
+
+    ACTION = UNIT_INSTALL_ACTION
+
+
+class UnitInstallScheduleResourceView(ConsumerUnitActionScheduleResourceView):
+    """
+    View for scheduled content install resource  on the consumer.
+    """
+
+    ACTION = UNIT_INSTALL_ACTION
+
+
+class UnitUpdateScheduleView(ConsumerUnitActionScheduleView):
+    """
+    View for scheduled content update on the consumer.
+    """
+
+    ACTION = UNIT_UPDATE_ACTION
+
+
+class UnitUpdateScheduleResourceView(ConsumerUnitActionScheduleResourceView):
+    """
+    View for scheduled content update resource on the consumer.
+    """
+
+    ACTION = UNIT_UPDATE_ACTION
+
+
+class UnitUninstallScheduleView(ConsumerUnitActionScheduleView):
+    """
+    View for scheduled content uninstall on the consumer.
+    """
+
+    ACTION = UNIT_UNINSTALL_ACTION
+
+
+class UnitUninstallScheduleResourceView(ConsumerUnitActionScheduleResourceView):
+    """
+    View for scheduled content uninstall resource on the consumer.
+    """
+
+    ACTION = UNIT_UNINSTALL_ACTION
+
+
+def add_link(consumer):
```

Comment:
These helpers are being "covered" by view tests but they should have their own tests too.